src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -7449,10 +7449,10 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         }
         for (size_t i=0; i<n_values; i++) {
             if ( is_string && !ASRUtils::is_integer(*expr_type(m_values[i])) ) {
-                throw CodeGenError("Only integer type is "
-                    "supported for string write(..) for now");
+                fmt.push_back("%s");
+                args.push_back(sep);
             }
-            if (i != 0 && !is_string) {
+            else if (i != 0 && !is_string) {
```

Comment:
Can you check if the following suggestion works?
```suggestion
            if (i != 0 && !is_string) {
```
I think the function `compute_fmt_specifier_and_arg`, do the same thing.